bluepysnap/simulation.py

```diff
@@ -125,8 +136,17 @@ def simulator(self):
     @cached_property
     def node_sets(self):
         """Returns the NodeSets object bound to the simulation."""
-        path = self.to_libsonata.node_sets_file
-        return NodeSets.from_file(path) if path else NodeSets.from_dict({})
+        try:
+            path = self.circuit.to_libsonata.node_sets_path
+        except BluepySnapError:
+            path = ""
+
+        node_sets = NodeSets.from_file(path) if path else NodeSets.from_dict({})
+
+        if path := self.to_libsonata.node_sets_file:
+            _warn_on_overwritten_node_sets(node_sets.update(NodeSets.from_file(path)))
```

Comment:
For clarity I would split this in 2 lines: 1 to update the node_sets, 2 to print the warning
```suggestion
            overwritten = node_sets.update(NodeSets.from_file(path))
            _warn_on_overwritten_node_sets(overwritten)
```

Reply:
Done